starters/qwik-graphql-tailwind/src/components/data-fetching/data-fetching.tsx

```diff
@@ -14,44 +14,69 @@ interface HelloResponse {
 }
 
 export const DataFetching = component$(() => {
-  const greetingResource = useResource$<HelloResponse>(({ cleanup }) => {
+  const store = useStore({
+    greeting: 'there',
+  });
+
+  const greetingResource = useResource$<HelloResponse>(({ track, cleanup }) => {
+    // Use `track` to trigger re-run of the the data fetching function.
+    track(() => store.greeting);
     // The `cleanup` function will be called when the function re-runs and the `AbortController` will abort the previous request.
     const abortController = new AbortController();
     cleanup(() => abortController.abort());
 
     // Fetch the the greeting and return Promise that resolves to the greeting.
-    return fetchGreeting('from This Dot Labs!', abortController);
+    return fetchGreeting(store.greeting, abortController);
   });
 
   return (
-    <div className="flex w-full text-xl justify-center">
-      <Resource
-        value={greetingResource}
-        onPending={() => <div className="text-left grow animate-pulse bg-gray-200 rounded-md">Loading...</div>}
-        onRejected={(error) => (
-          <div
-            className="grow border border-solid border-red-300 rounded bg-red-100 p-4 text-center text-red-500"
-            role="error-message"
-          >
-            Error: {error.message}
-          </div>
-        )}
-        onResolved={({ data }) => (
-          <>
-            <div className="mr-4">Message:</div>
-            <div className="text-left" role="display-message">
-              {data.hello}
+    <div>
+      <div class="text-center pb-8">
+        <input
+          type="text"
+          value={store.greeting}
+          placeholder="Who should the server greet?"
+          onInput$={(ev) => {
+            setTimeout(() => {
+              store.greeting = (ev.target as HTMLInputElement).value;
+            }, 1000);
```

Comment:
what's the essence of the setTimeout?

Reply:
just implementing a delay API call, to reduce errors shown, because of multiple API calls which abort the previous one on input changes.